tests/unit/test_files.py

```diff
@@ -0,0 +1,101 @@
+import io
+
+import pytest
+from west import files
+from west import settings
+
+
+@pytest.mark.parametrize("isdir,expected", [(True, ["15.0"]), (False, [])])
+def test_list_dirs(mocker, isdir, expected):
+    mocker.patch("os.listdir", return_value=["15.0"])
+    mocker.patch("os.path.isdir", return_value=isdir)
+
+    dirs = files.list_dirs("tests/test_data")
+
+    assert dirs == expected
+
+
+@pytest.mark.parametrize("isfile,expected", [(True, ["file.sql"]), (False, [])])
+def test_list_files(mocker, isfile, expected):
+    mocker.patch("os.listdir", return_value=["file.sql"])
+    mocker.patch("os.path.isfile", return_value=isfile)
+
+    values = files.list_files("tests/test_data/sql/fixtures")
+
+    assert values == expected
```

Comment:
Why using mocks instead of using our fixtures for those tests?

Reply:
We wanted to avoid any io operation in our unit test suite.